src/main/java/frc/robot/subsystems/ClimbSubsystem.java

```diff
@@ -16,49 +21,216 @@
+
+    climberTab = Shuffleboard.getTab("ClimbBase");
+    sbClimbingMode =
+        climberTab
+            .add("Manual Mode Enable", false)
+            .withSize(2, 2)
+            .withPosition(5, 1)
+            .withWidget(BuiltInWidgets.kToggleSwitch)
+            .getEntry();
+
+    sbClimberSpeedInput =
+        climberTab.add("Climber Speed input", 0).withSize(2, 2).withPosition(5, 0).getEntry();
+
+    sbClimbEnabbled =
+        climberTab.add("Climb Eanbled", false).withSize(2, 2).withPosition(0, 0).getEntry();
+    sbclimberpositionOne =
+        climberTab.add("Climber position", 0).withSize(2, 2).withPosition(2, 0).getEntry();
+    sbclimberspeedOne =
+        climberTab.add("Climber Current Speed 1", 0).withSize(2, 2).withPosition(4, 0).getEntry();
+    sbclimberheightOne =
+        climberTab
+            .add("Climber targetted height 1", 0)
+            .withSize(2, 2)
+            .withPosition(6, 0)
+            .getEntry();
+    sbclimbHeightOne =
+        climberTab.add("Climb Hight 1", 0).withSize(2, 2).withPosition(8, 0).getEntry();
+    sbclimbHeightTwo =
+        climberTab.add("Climb Hight 2", 0).withSize(2, 2).withPosition(8, 0).getEntry();
 
-    m_limitSwitch = new DigitalInput(Constants.LimitSwitchChannel);
+    sbclimberheightTwo =
+        climberTab
+            .add("Climber targetted height 2", 0)
+            .withSize(2, 2)
+            .withPosition(8, 0)
+            .getEntry();
+    sbclimberspeedTwo =
+        climberTab.add("Climber Current Speed 2", 0).withSize(2, 2).withPosition(0, 2).getEntry();
+
+    sbclimbHeightImputOne =
+        climberTab.add("Climber Imput One", 0).withSize(2, 2).withPosition(2, 2).getEntry();
+    sbclimbHeightImputTwo =
+        climberTab.add("Climber Imput Two", 0).withSize(2, 2).withPosition(4, 2).getEntry();
+    sbclimberpositionTwo =
+        climberTab.add("Climber position 2", 0).withSize(2, 2).withPosition(2, 4).getEntry();
+
+    m_climbMotorControllerOne.getPID().setP(Constants.climbRightPID[0]);
+    m_climbMotorControllerOne.getPID().setI(Constants.climbRightPID[1]);
+    m_climbMotorControllerOne.getPID().setD(Constants.climbRightPID[2]);
+
+    m_climbMotorControllerTwo.getPID().setP(Constants.climbLeftPID[0]);
+    m_climbMotorControllerTwo.getPID().setI(Constants.climbLeftPID[1]);
+    m_climbMotorControllerTwo.getPID().setD(Constants.climbLeftPID[2]);
   }
 
-  public void toggleClimbEnable() {
-    toggleClimb = !toggleClimb;
+  public void climbEnabbledEnable() {
+    climbEnabbled = !climbEnabbled;
+    sbClimbEnabbled.setBoolean(climbEnabbled);
+  }
+
+  public boolean getclimbingmode() {
+    return sbClimbingMode.getBoolean(false);
+  }
+
+  public void RunManual() {
+    if (climbEnabbled
+    /** && !m_limitSwitch.get() */
+    ) {
+
+      joystickAxis = -m_climbJoystick.getRawAxis(Constants.leftJoystickY);
+      if (joystickAxis > 0.1 || joystickAxis < -0.1) {
+        if (joystickAxis > 0) {
+          if (climbHeightOne <= 20.0) {
```

Comment:
climbHeightOne and climbHeightTwo are not needed in RunManual

Reply:
if are gone now